src/Race.js

```diff
@@ -0,0 +1,80 @@
+import {
+  getKeyArrayHasTargetValueInMap,
+  getMapFilledZero,
+  getMaxValueInMap,
+  getRepeatedString,
+  pickNumberInRange,
+  print,
+} from './lib/utils.js';
+
+class Race {
+  static #MIN_RANDOM = 0;
+
+  static #MAX_RANDOM = 9;
+
+  static #MOVE_FORWARD_THRESHOLD = 4;
+
+  static #TRACE_CHARACTER = '-';
+
+  static #FORWARD_STEP = 1;
+
+  #carArray;
+
+  #tryCount;
+
+  #carTraceMap;
+
+  constructor(carArray, tryCount) {
+    this.#carArray = carArray;
+    this.#tryCount = tryCount;
+
+    this.#carTraceMap = getMapFilledZero(this.#carArray);
+  }
+
+  run() {
+    for (let round = 0; round < this.#tryCount; round += 1) {
+      this.#runOneRound();
+    }
+    return this.#getWinnerCarArray;
+  }
+
+  #runOneRound() {
+    this.#carArray.forEach(car => {
+      const newPosition = Race.#getNewPosition(this.#carTraceMap.get(car));
+      this.#carTraceMap.set(car, newPosition);
+      Race.#printCarPosition(car, newPosition);
+    });
+  }
+
+  static #getNewPosition(currentPosition) {
+    const randomNum = pickNumberInRange(Race.#MIN_RANDOM, Race.#MAX_RANDOM);
+    const isMoveForward = Race.#getIsMoveForward(randomNum);
+
+    if (isMoveForward) return currentPosition + Race.#FORWARD_STEP;
+    return currentPosition;
+  }
+
+  static #getIsMoveForward(num) {
+    return num >= Race.#MOVE_FORWARD_THRESHOLD;
+  }
```

Comment:
```suggestion
  static #getNewPosition(currentPosition) {
    const randomNum = pickNumberInRange(Race.#MIN_RANDOM, Race.#MAX_RANDOM);

    if (this.#isMoveForward(randomNum)) return currentPosition + Race.#FORWARD_STEP;
    return currentPosition;
  }

  static #isMoveForward(num) {
    return num >= Race.#MOVE_FORWARD_THRESHOLD;
  }
```

isMoveForward의 결과를 얻지 말고 인라인으로 사용해도 좋을 거 같아요!

Reply:
이 방식도 상당히 더 간결한 것 같네요 !! 감사합니다.